### ISSUE TITLE
Allow setting label value and layer index for segment

### ISSUE THREAD

**Issue by lassoan** [Slicer/Slicer#4886]
To be compatible with external software that can only identify segments based on label value, we need to be able to give control to users to set label and layer value.

Since users may strongly rely on this association, there should be an option to lock label and layer values. Certain operations (such as having overlapping segments but have segments on the same layer) may not be possible with these locks, so a mechanism is needed to detect when this occurs and display a meaningful error message.

See https://discourse.slicer.org/t/how-to-change-segmentid-of-existing-segmentations/11254/4 for a related user request.

**Comment by anagrammarian** [comment]
Has this happened? I need to scale the value of a label by a prespecified number and add it to the image to "burn" in that label in the image (if the image intensity is 1-1000 and I add 1, it won't work so I have to scale the label by 500 or so and then add using Add Scalar volumes). I do this using fslmath which is a pain as I have to do that on a different machine. So ability to scale or change label value will be very cool.

**Comment by lassoan** [comment]
Yes, setting of label values for segments is implemented. You can specify label values in a color table. See details here: https://slicer.readthedocs.io/en/latest/user_guide/modules/segmentations.html#export-segmentation-to-labelmap-volume-file

We have not encountered any software in the past several years that could store a segmentation with overlapping labels in a single file (the very few that supports this do that by saving each label in a separate file), therefore we don't need to be able to set the layer for a segment for now.
